base/refpointer.jl

```diff
@@ -13,11 +13,20 @@ Creation of a `Ref` to a value `x` of type `T` is usually written `Ref(x)`.
 Additionally, for creating interior pointers to containers (such as Array or Ptr),
 it can be written `Ref(a, i)` for creating a reference to the `i`-th element of `a`.
 
-When passed as a `ccall` argument (either as a `Ptr` or `Ref` type), a `Ref` object will be
-converted to a native pointer to the data it references.
+`Ref{T}()` creates a reference to a value of type `T` without initialization.
+For a bitstype `T`, the value will be whatever currently resides in the memory
+allocated. For a non-bitstype `T`, the value will be `undef` and will result in
+an error, "UndefRefError: access to undefined reference" if the Ref is
+deferenced.
 
-There is no invalid (NULL) `Ref` in Julia, but a `C_NULL` instance of `Ptr` can be passed to
-a `ccall` Ref argument.
+To check if a `Ref` refers to [`undef`](@ref), use [`isassigned(ref::RefValue)`](@ref).
```

Comment:
I highly suspect that this will link to `undef`, which is wrong. It's a missing reference, it's not refering to any object. A reference to `undef` is `Ref(undef)`.

Reply:
I just say "undefined reference" now in 94edfaa